### PR TITLE
feat: add missing iOS Screen recording options

### LINKED ISSUE #400: Screen recording options not implemented?
## Description

I am experimenting with screen recording functionality. According to [this](http://appium.io/docs/en/commands/device/recording-screen/start-recording-screen/) specification I should be able to specify multiple parameters but it seems that some of them are not implemented in appium dotnet driver. For iOS missing implementations: options.videoFps, options.videoScale. Are these going to be implemented in the near feature?

## Environment

* .NET client build version: 4.1.1
* Appium server version: 1.15.1
* Desktop OS/version used to run Appium if necessary: macOS Catalina 10.15.2
* Node.js version: 12.14.1
* Mobile platform/version under test: iPhone X, iOS 13.3

## Code To Reproduce Issue

```
Driver.Instance.StartRecordingScreen(
             IOSStartScreenRecordingOptions.GetIosStartScreenRecordingOptions()
                        .WithVideoQuality(IOSStartScreenRecordingOptions.VideoQuality.MEDIUM)
                        .WithTimeLimit(TimeSpan.FromMinutes(5))
                        ...);
```

### LINKED ISSUE #404: Cannot record screen on iOS devices when video type is specified
## Description

I am trying to record a test execution on real iOS device. While experimenting with recording settings I found that if I specify any video type - the record will fail. If I do not specify any recording video type (then it defaults to mp4) it will succeed.

## Environment

* .NET client build version: 4.1.1
* Appium server version: 1.15.1
* Desktop OS/version used to run Appium if necessary: macOS Catalina 10.15.2
* Node.js version: 12.14.1
* Mobile platform/version under test: real device - iPhone X, iOS 13.3

## Details

Try specifying mp4:
```
Driver.Instance.StartRecordingScreen(
    IOSStartScreenRecordingOptions.GetIosStartScreenRecordingOptions()
        .WithVideoType(IOSStartScreenRecordingOptions.VideoType.MP4));
```
Error in Appium:
```
2020-04-20 12:34:42:037 [ffmpeg] ffmpeg version 4.2.2 Copyright (c) 2000-2019 the FFmpeg developers
2020-04-20 12:34:42:038 [ffmpeg]   built with Apple clang version 11.0.0 (clang-1100.0.33.17)
2020-04-20 12:34:42:038 [ffmpeg]   configuration: --prefix=/usr/local/Cellar/ffmpeg/4.2.2_2 --enable-shared --enable-pthreads --enable-version3 --enable-avresample --cc=clang --host-cflags= --host-ldflags= --enable-ffplay --enable-gnutls --enable-gpl --enable-libaom --enable-libbluray --enable-libmp3lame --enable-libopus --enable-librubberband --enable-libsnappy --enable-libtesseract --enable-libtheora --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-lzma --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librtmp --enable-libspeex --enable-libsoxr --enable-videotoolbox --disable-libjack --disable-indev=jack
2020-04-20 12:34:42:038 [ffmpeg] 
2020-04-20 12:34:42:038 [ffmpeg]   libavutil      56. 31.100 / 56. 31.100
2020-04-20 12:34:42:039 [ffmpeg]   libavcodec     58. 54.100 / 58. 54.100
2020-04-20 12:34:42:039 [ffmpeg]   libavformat    58. 29.100 / 58. 29.100
2020-04-20 12:34:42:039 [ffmpeg]   libavdevice    58.  8.100 / 58.  8.100
2020-04-20 12:34:42:039 [ffmpeg]   libavfilter     7. 57.100 /  7. 57.100
2020-04-20 12:34:42:039 [ffmpeg]   libavresample   4.  0.  0 /  4.  0.  0
2020-04-20 12:34:42:039 [ffmpeg]   libswscale      5.  5.100 /  5.  5.100
2020-04-20 12:34:42:039 [ffmpeg]   libswresample   3.  5.100 /  3.  5.100
2020-04-20 12:34:42:040 [ffmpeg]   libpostproc    55.  5.100 / 55.  5.100
2020-04-20 12:34:42:040 [ffmpeg] 
2020-04-20 12:34:42:232 [ffmpeg] Input #0, mjpeg, from 'http://localhost:9100':
2020-04-20 12:34:42:232 [ffmpeg]   Duration: N/A, bitrate: N/A
2020-04-20 12:34:42:232 [ffmpeg] 
2020-04-20 12:34:42:233 [ffmpeg]     Stream #0:0: Video: mjpeg (Baseline), yuvj420p(pc, bt470bg/unknown/unknown), 1125x2436 [SAR 216:216 DAR 375:812], 25 tbr, 1200k tbn, 25 tbc
2020-04-20 12:34:42:233 [ffmpeg] Unknown encoder 'mp4'
2020-04-20 12:34:42:233 [ffmpeg] 
2020-04-20 12:34:42:235 [iProxy@36fad...] Error: read ECONNRESET
2020-04-20 12:34:42:235 [iProxy@36fad...]     at TCP.onStreamRead (internal/stream_base_commons.js:200:27)
2020-04-20 12:34:47:040 [XCUITest] Screen capture process did not start within 5000ms. Continuing anyway
2020-04-20 12:34:47:040 [DevCon Factory] Releasing connections for 36fadaa9cb9516ba18bd0a7e764b8369fef568b6 device on 9100 port number
2020-04-20 12:34:47:041 [DevCon Factory] Found cached connections to release: ["36fadaa9cb9516ba18bd0a7e764b8369fef568b6:9100"]
2020-04-20 12:34:47:041 [DevCon Factory] Releasing the listener for '36fadaa9cb9516ba18bd0a7e764b8369fef568b6:9100'
2020-04-20 12:34:47:042 [DevCon Factory] Cached connections count: 1
2020-04-20 12:34:47:053 [W3C (90f39573)] Encountered internal error running command: Error: The screen capture process 'ffmpeg' died unexpectedly. Check server logs for more details
2020-04-20 12:34:47:053 [W3C (90f39573)]     at ScreenRecorder.start (/Users/bc9667/.nvm/versions/node/v12.14.1/lib/node_modules/appium/node_modules/appium-xcuitest-driver/lib/commands/recordscreen.js:114:13)
2020-04-20 12:34:47:062 [HTTP] <-- POST /wd/hub/session/90f39573-e3a0-4eb2-a586-b178bf710ff6/appium/start_recording_screen 500 5244 ms - 737
```
Try specifying H264:
```
Driver.Instance.StartRecordingScreen(
    IOSStartScreenRecordingOptions.GetIosStartScreenRecordingOptions()
        .WithVideoType(IOSStartScreenRecordingOptions.VideoType.H264));
```
Error in Appium:
```
2020-04-20 12:15:02:108 [ffmpeg] ffmpeg version 4.2.2 Copyright (c) 2000-2019 the FFmpeg developers
2020-04-20 12:15:02:108 [ffmpeg]   built with Apple clang version 11.0.0 (clang-1100.0.33.17)
2020-04-20 12:15:02:109 [ffmpeg]   configuration: --prefix=/usr/local/Cellar/ffmpeg/4.2.2_2 --enable-shared --enable-pthreads --enable-version3 --enable-avresample --cc=clang --host-cflags= --host-ldflags= --enable-ffplay --enable-gnutls --enable-gpl --enable-libaom --enable-libbluray --enable-libmp3lame --enable-libopus --enable-librubberband --enable-libsnappy --enable-libtesseract --enable-libtheora --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-lzma --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librtmp --enable-libspeex --enable-libsoxr --enable-videotoolbox --disable-libjack --disable-indev=jack
2020-04-20 12:15:02:109 [ffmpeg] 
2020-04-20 12:15:02:109 [ffmpeg]   libavutil      56. 31.100 / 56. 31.100
2020-04-20 12:15:02:109 [ffmpeg]   libavcodec     58. 54.100 / 58. 54.100
2020-04-20 12:15:02:110 [ffmpeg]   libavformat    58. 29.100 / 58. 29.100
2020-04-20 12:15:02:110 [ffmpeg]   libavdevice    58.  8.100 / 58.  8.100
2020-04-20 12:15:02:110 [ffmpeg]   libavfilter     7. 57.100 /  7. 57.100
2020-04-20 12:15:02:110 [ffmpeg]   libavresample   4.  0.  0 /  4.  0.  0
2020-04-20 12:15:02:110 [ffmpeg]   libswscale      5.  5.100 /  5.  5.100
2020-04-20 12:15:02:110 [ffmpeg]   libswresample   3.  5.100 /  3.  5.100
2020-04-20 12:15:02:111 [ffmpeg]   libpostproc    55.  5.100 / 55.  5.100
2020-04-20 12:15:02:111 [ffmpeg] 
2020-04-20 12:15:02:304 [ffmpeg] Input #0, mjpeg, from 'http://localhost:9100':
2020-04-20 12:15:02:305 [ffmpeg]   Duration: N/A, bitrate: N/A
2020-04-20 12:15:02:305 [ffmpeg] 
2020-04-20 12:15:02:305 [ffmpeg]     Stream #0:0: Video: mjpeg (Baseline), yuvj420p(pc, bt470bg/unknown/unknown), 1125x2436 [SAR 216:216 DAR 375:812], 25 tbr, 1200k tbn, 25 tbc
2020-04-20 12:15:02:305 [ffmpeg] Stream mapping:
2020-04-20 12:15:02:306 [ffmpeg]   Stream #0:0 -> #0:0 (mjpeg (native) -> h264 (libx264))
2020-04-20 12:15:02:306 [ffmpeg] Press [q] to stop, [?] for help
2020-04-20 12:15:02:306 [ffmpeg] 
2020-04-20 12:15:02:316 [ffmpeg] [libx264 @ 0x7f9ed3804600] width not divisible by 2 (1125x2436)
2020-04-20 12:15:02:317 [ffmpeg] 
2020-04-20 12:15:02:317 [ffmpeg] Error initializing output stream 0:0 -- Error while opening encoder for output stream #0:0 - maybe incorrect parameters such as bit_rate, rate, width or height
2020-04-20 12:15:02:317 [ffmpeg] Conversion failed!
2020-04-20 12:15:02:317 [ffmpeg] 
2020-04-20 12:15:02:318 [iProxy@36fad...] Error: read ECONNRESET
2020-04-20 12:15:02:318 [iProxy@36fad...]     at TCP.onStreamRead (internal/stream_base_commons.js:200:27)
2020-04-20 12:15:07:093 [XCUITest] Screen capture process did not start within 5000ms. Continuing anyway
2020-04-20 12:15:07:094 [DevCon Factory] Releasing connections for 36fadaa9cb9516ba18bd0a7e764b8369fef568b6 device on 9100 port number
2020-04-20 12:15:07:094 [DevCon Factory] Found cached connections to release: ["36fadaa9cb9516ba18bd0a7e764b8369fef568b6:9100"]
2020-04-20 12:15:07:095 [DevCon Factory] Releasing the listener for '36fadaa9cb9516ba18bd0a7e764b8369fef568b6:9100'
2020-04-20 12:15:07:096 [DevCon Factory] Cached connections count: 1
2020-04-20 12:15:07:110 [W3C (e309ece2)] Encountered internal error running command: Error: The screen capture process 'ffmpeg' died unexpectedly. Check server logs for more details
2020-04-20 12:15:07:110 [W3C (e309ece2)]     at ScreenRecorder.start (/Users/bc9667/.nvm/versions/node/v12.14.1/lib/node_modules/appium/node_modules/appium-xcuitest-driver/lib/commands/recordscreen.js:114:13)
2020-04-20 12:15:07:119 [HTTP] <-- POST /wd/hub/session/e309ece2-c48a-4c70-a291-cff8929e55a9/appium/start_recording_screen 500 5230 ms - 737
```
Try specifying FMP4:
```
Driver.Instance.StartRecordingScreen(
    IOSStartScreenRecordingOptions.GetIosStartScreenRecordingOptions()
        .WithVideoType(IOSStartScreenRecordingOptions.VideoType.FMP4));
```
Error in Appium:
```
2020-04-20 12:37:46:393 [ffmpeg] ffmpeg version 4.2.2 Copyright (c) 2000-2019 the FFmpeg developers
2020-04-20 12:37:46:393 [ffmpeg]   built with Apple clang version 11.0.0 (clang-1100.0.33.17)
2020-04-20 12:37:46:393 [ffmpeg]   configuration: --prefix=/usr/local/Cellar/ffmpeg/4.2.2_2 --enable-shared --enable-pthreads --enable-version3 --enable-avresample --cc=clang --host-cflags= --host-ldflags= --enable-ffplay --enable-gnutls --enable-gpl --enable-libaom --enable-libbluray --enable-libmp3lame --enable-libopus --enable-librubberband --enable-libsnappy --enable-libtesseract --enable-libtheora --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-lzma --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librtmp --enable-libspeex --enable-libsoxr --enable-videotoolbox --disable-libjack --disable-indev=jack
2020-04-20 12:37:46:393 [ffmpeg] 
2020-04-20 12:37:46:394 [ffmpeg]   libavutil      56. 31.100 / 56. 31.100
2020-04-20 12:37:46:394 [ffmpeg]   libavcodec     58. 54.100 / 58. 54.100
2020-04-20 12:37:46:394 [ffmpeg]   libavformat    58. 29.100 / 58. 29.100
2020-04-20 12:37:46:394 [ffmpeg]   libavdevice    58.  8.100 / 58.  8.100
2020-04-20 12:37:46:394 [ffmpeg]   libavfilter     7. 57.100 /  7. 57.100
2020-04-20 12:37:46:394 [ffmpeg]   libavresample   4.  0.  0 /  4.  0.  0
2020-04-20 12:37:46:394 [ffmpeg]   libswscale      5.  5.100 /  5.  5.100
2020-04-20 12:37:46:394 [ffmpeg]   libswresample   3.  5.100 /  3.  5.100
2020-04-20 12:37:46:395 [ffmpeg]   libpostproc    55.  5.100 / 55.  5.100
2020-04-20 12:37:46:395 [ffmpeg] 
2020-04-20 12:37:46:644 [ffmpeg] Input #0, mjpeg, from 'http://localhost:9100':
2020-04-20 12:37:46:644 [ffmpeg]   Duration: N/A, bitrate: N/A
2020-04-20 12:37:46:644 [ffmpeg] 
2020-04-20 12:37:46:645 [ffmpeg]     Stream #0:0: Video: mjpeg (Baseline), yuvj420p(pc, bt470bg/unknown/unknown), 1125x2436 [SAR 216:216 DAR 375:812], 25 tbr, 1200k tbn, 25 tbc
2020-04-20 12:37:46:645 [ffmpeg] Unknown encoder 'fmp4'
2020-04-20 12:37:46:645 [ffmpeg] 
2020-04-20 12:37:46:646 [iProxy@36fad...] Error: read ECONNRESET
2020-04-20 12:37:46:646 [iProxy@36fad...]     at TCP.onStreamRead (internal/stream_base_commons.js:200:27)
2020-04-20 12:37:51:446 [XCUITest] Screen capture process did not start within 5000ms. Continuing anyway
2020-04-20 12:37:51:446 [DevCon Factory] Releasing connections for 36fadaa9cb9516ba18bd0a7e764b8369fef568b6 device on 9100 port number
2020-04-20 12:37:51:446 [DevCon Factory] Found cached connections to release: ["36fadaa9cb9516ba18bd0a7e764b8369fef568b6:9100"]
2020-04-20 12:37:51:447 [DevCon Factory] Releasing the listener for '36fadaa9cb9516ba18bd0a7e764b8369fef568b6:9100'
2020-04-20 12:37:51:448 [DevCon Factory] Cached connections count: 1
2020-04-20 12:37:51:458 [W3C (dedc9607)] Encountered internal error running command: Error: The screen capture process 'ffmpeg' died unexpectedly. Check server logs for more details
2020-04-20 12:37:51:458 [W3C (dedc9607)]     at ScreenRecorder.start (/Users/bc9667/.nvm/versions/node/v12.14.1/lib/node_modules/appium/node_modules/appium-xcuitest-driver/lib/commands/recordscreen.js:114:13)
2020-04-20 12:37:51:467 [HTTP] <-- POST /wd/hub/session/dedc9607-9b06-48ad-8011-7feda39c53cf/appium/start_recording_screen 500 5257 ms - 737
```

### PR DESCRIPTION
**Pull request by akinsolb**
## Change list

Please provide briefly described change list which are you going to propose. 
 
## Types of changes

What types of changes are you proposing/introducing to .NET client?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] New feature (non-breaking change which adds value to the project)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Documentation
- [x] Have you proposed a file change/ PR with appium to update documentation? 
## Integration tests
- [x] Have you provided integration tests to pass against the beta version of appium? (for Bugfix or New feature)

## Details
Added an overload for videoType option
Added the following options for iOS: videoScale, videoFps, videoPixelFormat, videoFilters
closes #400 
closes #404 

Please provide more details about changes if it is necessary. If there are new features you can provide code samples which show the way they
work and possible use cases. Also you can create [gists](https://gist.github.com) with pasted C# code samples or put them here using markdown. 
About markdown please read [Mastering markdown](https://guides.github.com/features/mastering-markdown/) and [Writing on GitHub](https://help.github.com/categories/writing-on-github/)

### CODE REVIEW

File: src/Appium.Net/Appium/ScreenRecording/BaseStartScreenRecordingOptions.cs [hunk 1]
```diff
@@ -11,7 +13,8 @@ public abstract class BaseStartScreenRecordingOptions<T> : BaseScreenRecordingOp
         /// <returns>self instance for chaining.</returns>
         public T WithTimeLimit(TimeSpan timeLimit)
         {
-            Parameters["timeLimit"] = (int) timeLimit.TotalSeconds;
+            Parameters["timeLimit"] = timeLimit.TotalSeconds
+                .ToString(CultureInfo.InvariantCulture);
```
Comment (mykola-mokhnach) [inline src/Appium.Net/Appium/ScreenRecording/BaseStartScreenRecordingOptions.cs:17]: is it necessary to convert it to string?

Comment (akinsolb) [reply src/Appium.Net/Appium/ScreenRecording/BaseStartScreenRecordingOptions.cs:17]: Not sure if it's necessary but all the documentation I saw stated string.

Comment (mykola-mokhnach) [reply src/Appium.Net/Appium/ScreenRecording/BaseStartScreenRecordingOptions.cs:17]: I assume the server supports both string and numbers, but it makes more sense to keep it as int for me

Comment (akinsolb) [reply src/Appium.Net/Appium/ScreenRecording/BaseStartScreenRecordingOptions.cs]: Reverted

**Pull request merged**